### ISSUE TITLE
sidePanel won't re-open after switch back

### ISSUE THREAD

**Issue by Xheldon** [microsoft/MicrosoftEdge-Extensions#142]
My extension has the same problem. After using the [official example code](https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/developer-guide/sidebar#enable-a-sidebar-for-a-specific-site-only), once the user switches to another tab, the sidebar disappears and it does not automatically open again when switching back.

So when I tried to reopen it using this interface, I got the following error（only in Edge):

```bash
Uncaught (in promise) Error: `sidePanel.open()` may only be called in response to a user gesture.
```

Here is my code in background.js:

```js
chrome.tabs.onActivated.addListener(async (activeInfo) => {
    const tabId = activeInfo.tabId;
    const tab = await chrome.tabs.get(tabId);
    if (!tab.url) return;
    const url = new URL(tab.url);
    if (url.origin === 'https://www.notion.so') {
        await chrome.sidePanel.setOptions({
            tabId,
            path: 'sidepanel.html',
            enabled: true
        });
        // EDGE-ONLY: just edge browser has this problem, so I add this code blow, but I got the error and it's not work
        await chrome.sidePanel.open({
            tabId,
        });
    } else {
        // Disables the side panel on all other sites
        await chrome.sidePanel.setOptions({
            tabId,
            enabled: false
        });
    }
});
```

Please note, on Chrome, even without that piece of EDGE-ONLY code, it still works as expected: the sidebar automatically closes when switching to non-notion.so tabs, and the sidebar remains in its pre-switch open state when switching back to a notion.so tab. 
However, on Edge, the sidebar will close when switching to a non-notion.so tab (as expected), but when switching back to a notion.so tab, the sidebar will not remain open, and it has to be manually reopened at this point.
Even if I add that EDGE-ONLY code, it still doesn't work and also throws an error.

@Gurjeet-msft  Any idea？

_Originally posted by @Xheldon in https://github.com/microsoft/MicrosoftEdge-Extensions/discussions/95#discussioncomment-8884775_

**Comment by Xheldon** [comment]
@Manikanth2109  YES, the problem still exists. Version is: 125.0.2535.85 (Production) (arm64).

**Comment by Manikanth2109** [comment]
Thank you @Xheldon for the confirmation. We'll contact the right team responsible and get this fixed as soon as possible!

**Comment by zoulou00** [comment]
Hello,
Do you have any updates on the timeline for the resolution? 😊

**Comment by ManikanthMSFT** [comment]
@zoulou00, thank you for reaching out. We understand your eagerness for an update. Please be assured that our team is diligently working on the issue. 

Once we have more information or an update on the timeline for resolution, we will promptly write back to you. Your understanding and patience are appreciated.

**Comment by thryyy** [comment]
Hi @rishikramena-ms 

Thank you for your explanation. I now understand that the sidepanel is attached to the window rather than to each tab, which clarifies the behavior I was observing.

However, I'd like to point out that this behavior differs from Chrome, where the sidepanel reopens if it was already open for a specific tab. As an extension developer, one of the exciting aspects of Edge being based on Chromium was the ability to develop for both browsers without requiring browser-specific code. I'm concerned that these diverging behaviors might lead us back to the challenges we faced before this convergence.

Another difference I've noticed is that in Edge, the sidepanel width resets each time it's closed, whereas in Chrome, it's saved for the entire session until the browser is closed.

Thank you for considering this feedback.

**Comment by iNick** [comment]
I want to emphasize what @thryyy wrote above. The different design for how Edge handles side panels (shared for all tabs in a window) compared to Chrome (unique per tab) is also causing us a lot of headaches and extra work.

I also found this discussion pointing out the inconsistencies in the implementations of the sidePanel API: https://github.com/w3c/webextensions/issues/588

**Comment by sszczep** [comment]
Is there any work being done on that issue? It's a crucial issue for us since we would need to change the entire UX.

**Comment by Rahul-Bauri** [comment]
Hi @Xheldon @zoulou00 @thryyy @iNick @sszczep 

I am **Rahul** from the **Microsoft Edge Extensions Developer Support** team.

Please refer to #222 for further updates regarding this issue, as it is closely related. To keep all sidepanel-related discussions centralized, I’m closing this issue as a duplicate of #222. Feel free to share any suggestions or report any problems directly in #222. Thank you for your understanding!

**Comment by th3rdtim** [comment]
Not really familiar with Edge API, but could this be a case of 'user action' disappearing because  of awaiting promises hijacking the user action?
```ts
const tab = await chrome.tabs.get(tabId);
```
Awaiting promises has been known to hijack the user action.
